src/test/java/seedu/address/logic/commands/FilterCommandTest.java

```diff
@@ -0,0 +1,105 @@
+    private Model expectedModel = new ModelManager(TypicalFlashcards.getTypicalFlashcardDeck(), new UserPrefs());
+
+    @Test
+    public void equals() {
+        CategoryEqualsKeywordsPredicate firstPredicate =
+                new CategoryEqualsKeywordsPredicate(Collections.singletonList("first"));
+        CategoryEqualsKeywordsPredicate secondPredicate =
+                new CategoryEqualsKeywordsPredicate(Collections.singletonList("second"));
+
+        FilterCommand findFirstCommand = new FilterCommand(firstPredicate);
+        FilterCommand findSecondCommand = new FilterCommand(secondPredicate);
+
+        // same object -> returns true
+        assertTrue(findFirstCommand.equals(findFirstCommand));
+
+        // same values -> returns true
+        FilterCommand findFirstCommandCopy = new FilterCommand(firstPredicate);
+        assertTrue(findFirstCommand.equals(findFirstCommandCopy));
+
+        // different types -> returns false
+        assertFalse(findFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(findFirstCommand.equals(null));
+
+        // different person -> returns false
+        assertFalse(findFirstCommand.equals(findSecondCommand));
+    }
+
+    @Test
+    public void execute_zeroKeywords_noFlashcardFound() {
+        String expectedMessage = String.format(MESSAGE_FLASHCARDS_LISTED_OVERVIEW, 0);
+        CategoryEqualsKeywordsPredicate predicate = preparePredicate(" ");
+        FilterCommand command = new FilterCommand(predicate);
+        expectedModel.updateFilteredFlashcardList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Collections.emptyList(), model.getFilteredFlashcardList());
+    }
+
+    @Test
+    public void execute_multipleKeywords_flashcardFound() {
+        String expectedMessage = String.format(MESSAGE_FLASHCARDS_LISTED_OVERVIEW, 1);
+        CategoryEqualsKeywordsPredicate predicate = preparePredicate("c/SDLC");
+        FilterCommand command = new FilterCommand(predicate);
+        expectedModel.updateFilteredFlashcardList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(FLASHCARD_1), model.getFilteredFlashcardList());
+    }
+
+    @Test
+    public void execute_multipleKeywords_multipleFlashcardsFound() {
+        String expectedMessage = String.format(MESSAGE_FLASHCARDS_LISTED_OVERVIEW, 2);
+        CategoryEqualsKeywordsPredicate predicate = preparePredicate("c/SDLC c/Revision History");
+        FilterCommand command = new FilterCommand(predicate);
+        expectedModel.updateFilteredFlashcardList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(FLASHCARD_1, FLASHCARD_2), model.getFilteredFlashcardList());
+    }
+
+    @Test
+    public void execute_lowercaseKeyword() {
```

Comment:
Maybe can rename this to `execute_lowercaseKeyword_success`?

Reply:
Got it! Will do. Thanks